Wikipedia/Networking/Fetchers/SavedArticlesFetcher.h

```diff
@@ -17,14 +23,14 @@
 + (SavedArticlesFetcher*)sharedInstance;
 + (void)                 setSharedInstance:(SavedArticlesFetcher*)fetcher;
 
-@property (nonatomic, strong, readonly) MWKSavedPageList* savedPageList;
-@property (nonatomic, strong, readonly) MWKDataStore* dataStore;
-
-- (instancetype)initWithDataStore:(MWKDataStore*)dataStore;
+@property (nonatomic, strong) MWKSavedPageList* savedPageList;
 
 @property (nonatomic, weak) id<SavedArticlesFetcherDelegate> fetchFinishedDelegate;
 
-- (void)fetchSavedPageList:(MWKSavedPageList*)savedPageList;
+- (instancetype)initWithDataStore:(MWKDataStore*)dataStore;
```

Comment:
These 2 initializers seem very different. 
We should create an initializer in the way that we intend on using it in the app. 

if you are creating extra initializers for testing, as I have mentioned before, they should be in private headers and not in the public API. 

We should always strive for the most obvious well understood public APIs in the classes we create - these 2 initializers in particular make it very non-obvious which is the right one to use.